qiskit/compiler/assembler.py

```diff
@@ -8,43 +8,104 @@
                          QasmQobjInstruction, QasmQobjExperimentConfig, QasmQobjExperiment,
                          QasmQobjConfig, PulseQobjInstruction, PulseQobjExperimentConfig,
                          PulseQobjExperiment, PulseQobjConfig, QobjPulseLibrary)
 from qiskit.qobj.converters import PulseQobjConverter, LoConfigConverter
-from .run_config import RunConfig
 
 logger = logging.getLogger(__name__)
 
 
-def assemble_circuits(circuits, run_config=None, qobj_header=None, qobj_id=None):
+def assemble_circuits(circuits, qobj_id=None, qobj_header=None,
+                      shots=None, memory=None, max_credits=None,
+                      seed_simulator=None, run_config=None,
+                      config=None, seed=None):  # deprecated
     """Assembles a list of circuits into a qobj which can be run on the backend.
 
+    This function serializes the circuits to create Qobj "experiments", and
+    annotates the experiment payload with header and configurations.
+
     Args:
-        circuits (list[QuantumCircuits] or QuantumCircuit): circuits to assemble
-        run_config (RunConfig): RunConfig object
-        qobj_header (QobjHeader): header to pass to the results
-        qobj_id (int): identifier for the generated qobj
+        circuits (list[QuantumCircuits] or QuantumCircuit):
+            circuit(s) to assemble
+
+        qobj_id (str):
+            String identifier to annotate the Qobj
+
+        qobj_header (QobjHeader or dict):
+            User input that will be inserted in Qobj header, and will also be
+            copied to the corresponding Result header. Headers do not affect the run.
+
+        shots (int):
+            number of repetitions of each circuit, for sampling. Default: 2014
+
+        memory (bool):
+            if True, per-shot measurement bitstrings are returned as well
+            (provided the backend supports it). Default: False
+
+        max_credits (int):
+            maximum credits to spend on job. Default: 10
+
+        seed_simulator (int):
+            random seed to control sampling, for when backend is a simulator
+
+        config (dict):
+            DEPRECATED in 0.8: use run_config instead
+
+        run_config (RunConfig):
+            Qobj runtime configuration, containing some or all of the above options.
+            If any other option is explicitly set (e.g. shots), it
+            will override the run_config's.
+
+        seed (int):
+            DEPRECATED in 0.8: use ``seed_simulator`` kwarg instead
 
     Returns:
         QasmQobj: the Qobj to be run on the backends
     """
+    # deprecation matter
+    if config:
+        warnings.warn('config is not used anymore. Set all configs in '
+                      'run_config.', DeprecationWarning)
+        run_config = run_config or config
+    if seed:
+        warnings.warn('seed is deprecated in favor of seed_simulator.', DeprecationWarning)
+        seed_simulator = seed_simulator or seed
```

Comment:
Same as https://github.com/Qiskit/qiskit-terra/pull/2166/files#r277392245
```suggestion
        seed_simulator = seed or seed_simulator
```